packages/server/src/controllers/files.ts

```diff
@@ -2,58 +2,64 @@ import { Middleware } from 'koa'
 import ipfsClient, { globSource } from 'ipfs-http-client'
 import fs from 'fs'
 import { promisify } from 'util'
+import { BAD_REQUEST } from 'http-status-codes'
 
 import { ObjectionModels } from '@aragonone/ipfs-background-service-shared'
-import { FilesValidator } from '../helpers'
+import FilesValidator from '../helpers/files-validator'
 
 const { File } = ObjectionModels
 const ipfs = ipfsClient(process.env.IPFS_API_URL)
 const deleteTempFile = promisify(fs.unlink)
 
-const dummy_file = {
-  sizeBytes: 5712538,
-  extension: "pdf",    
-  encoding: "UTF-8",
-  createdAt: 1590322097,
-  owner: "0x...",
-  verified: false,
-  transactionHash: "0x...",
-  cid: "Qm..."
-}
-
 export default class FilesController {
   static create: Middleware = async (ctx) => {
     const { body: { owner }, file } = ctx.request
+    let cid = ''
     try {
-      FilesValidator.validateForCreate(ctx)
+      await FilesValidator.validateForCreate(ctx)
       const ipfsFile = await ipfs.add(globSource(file.path))
-      dummy_file.cid = ipfsFile.cid.toString()
-      await File.create('test')
+      cid = ipfsFile.cid.toString()
+      if (await File.exists({cid})) {
+        ctx.throw(BAD_REQUEST, { errors: [ { file: `File is already uploaded with cid ${cid}`} ] })
+      }
+      await File.create({
+        owner,
+        cid,
+        sizeBytes: file.size,
+        originalName: file.originalname,
+        encoding: file.encoding,
+        mimeType: file.mimetype
+      })
     }
     catch (err) {
       await deleteTempFile(file.path)
       throw err
     }
     await deleteTempFile(file.path)
-    ctx.body = dummy_file
+    ctx.body = await File.findOne({cid})
```

Comment:
Shouldn't these two lines above be inside the `try` clause?

Reply:
I just need that try catch to make sure I delete the temp file, I don't have it at this point. There is still general error handler as top middleware.